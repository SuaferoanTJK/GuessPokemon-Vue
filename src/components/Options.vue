<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
    showPokemon: { type: Boolean, required: true },
  },
};
</script>

<template>
  <div v-if="!showPokemon" class="mt-10 grid md:grid-cols-2 gap-4">
    <button
      v-for="pokemon in pokemons"
      :key="pokemon.id"
      @click="$emit('selection', pokemon.id)"
      class="rounded text-lg capitalize text-white hover:font-semibold bg-tertiary p-2 opacity-80 hover:opacity-100 cursor-pointer w-full shadow-md"
    >
      {{ pokemon.name }}
    </button>
  </div>
</template>
