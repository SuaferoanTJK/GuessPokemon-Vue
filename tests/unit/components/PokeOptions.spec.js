import { shallowMount } from "@vue/test-utils";
import { pokemons } from "../utils";
import Options from "@/components/Options";

describe("Options Unit Testing", () => {
  const props = {
    pokemons,
    showPokemon: false,
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Options, {
      props,
    });
  });
  test("snapshot must be matched", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("should display the 4 options", () => {
    const btns = wrapper.findAll("button");
    expect(btns.length).toBe(4);
    expect(btns[0].text()).toBe("bulbasaur");
    expect(btns[1].text()).toBe("ivysaur");
    expect(btns[2].text()).toBe("venusaur");
    expect(btns[3].text()).toBe("charmander");
  });
  test("on click the selection emitter must be triggered with the its params", () => {
    const [btn1, btn2, btn3, btn4] = wrapper.findAll("button");
    btn1.trigger("click");
    btn2.trigger("click");
    btn3.trigger("click");
    btn4.trigger("click");
    expect(wrapper.emitted("selection").length).toBe(4);
    expect(wrapper.emitted("selection")[0]).toEqual([1]);
    expect(wrapper.emitted("selection")[1]).toEqual([2]);
    expect(wrapper.emitted("selection")[2]).toEqual([3]);
    expect(wrapper.emitted("selection")[3]).toEqual([4]);
  });
});
