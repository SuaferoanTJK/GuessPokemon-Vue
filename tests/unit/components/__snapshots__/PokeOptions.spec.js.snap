// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Unit Testing snapshot must be matched 1`] = `<div class="mt-10 grid md:grid-cols-2 gap-4"><button class="rounded text-lg capitalize text-white hover:font-semibold bg-tertiary p-2 opacity-80 hover:opacity-100 cursor-pointer w-full shadow-md">bulbasaur</button><button class="rounded text-lg capitalize text-white hover:font-semibold bg-tertiary p-2 opacity-80 hover:opacity-100 cursor-pointer w-full shadow-md">ivysaur</button><button class="rounded text-lg capitalize text-white hover:font-semibold bg-tertiary p-2 opacity-80 hover:opacity-100 cursor-pointer w-full shadow-md">venusaur</button><button class="rounded text-lg capitalize text-white hover:font-semibold bg-tertiary p-2 opacity-80 hover:opacity-100 cursor-pointer w-full shadow-md">charmander</button></div>`;
